metricflow/sql/optimizer/rewriting_sub_query_reducer.py

```diff
@@ -216,6 +216,14 @@ def _current_node_can_be_reduced(self, node: SqlSelectStatementNode) -> bool:
         if len(node.join_descs) > 0:
             return False
 
+        # If the parent node defines CTEs, don't reduce for simplicity. It's possible to improve this by keeping track
```

Comment:
This might impact the output of the custom offset window query 🤔 but we can potentially address that later.